cli/util/options.d.ts

```diff
@@ -53,3 +56,9 @@ interface HelpOptions {
 
 /** Generates the help text for the specified configuration. */
 export function help(config: Config, options?: HelpOptions): string;
+
+/** Populates default values on a parsed options result. */
+export function addDefaults(config: Config, options: OptionSet): OptionSet;
+
+/** Merges two sets of options into one, preferring the newer set. */
+export function merge(config: Config, currentOptions: OptionSet, parentOptions: OptionSet): OptionSet;
```

Comment:
How can we use this in asc.js? I propose something like this. 

```ts
let opts = parse(argv, config);
let asconfig = getAsconfig(opts.config);
while (asconfig) {
  merge(config, opts, merge(config, 
  asconfig.targets[target] || {}, 
  asconfig.options || {});
  asconfig = asconfig.extends
    ? getAsconfig(asconfig.extends)
    : null;
}
```

Reply:
Seems good. But note that `parse` returns a full result while `merge` operations just on its `.options` and returns the new set.

Reply:
I didn't flesh everything out. Would I be responsible for implementing this algorithm?

Reply:
So far this is just the algorithm, so if you'd like to continue on your PR based on this, feel free to do so. If not, just let me know :)